inject-generator/src/main/java/io/avaje/inject/generator/MethodReader.java

```diff
@@ -526,7 +533,7 @@ boolean isLazy() {
   }
 
   boolean isUseProviderForSecondary() {
-    return secondary && !optionalType && !Util.isProvider(returnTypeRaw);
+    return (secondary || priority != null) && !optionalType && !Util.isProvider(returnTypeRaw);
```

Comment:
So if it's got a `@Priority` also wire as `Provider<T>` which has the effect of only instantiating when needed.

Needed by .. being the highest priority or BeanScope#list or injecting a List/Set/Map of that type.

Reply:
Indeed, the appearance of a priority annotation invites the possibility that a higher priority bean might exist